service_container/factories.rst

```diff
@@ -293,5 +293,138 @@ previous examples takes the ``templating`` service as an argument:
                 ->args([ref('templating')])
             ;
         };
+        
+Usage example
+-------------
+
+The following example is intended to show how to create and use a factory method in Symfony framework.
+Suppose you want to realize the factory method pattern for services, that describe two delivery methods - DHL and UPS.
+
+Services (subclasses) definition::
+
+    // src/Deliveries/DHL.php
+    namespace App\Deliveries;
+
+    class DHL
+    {
+        public $costLabel;
+    
+        public function cost() {
+            return 100;
+        }
+    }
+    
+    // src/Deliveries/UPS.php
+    namespace App\Deliveries;
+
+    class UPS
+    {
+        public $costLabel;
+    
+        public function cost() {
+            return 200;
+        }
+    }
+
+Factory definition::
+
+    // src/Factories/DeliveryFactory.php
+    namespace App\Factories;
+    
+    abstract class DeliveryFactory
+    {
+        public static function create($deliveryMethod)
+        {
+            $delivery = new $deliveryMethod;
+            $delivery->costLabel = 'Delivery cost is: ';
+    
+            return $delivery;
+        }
+        
+        abstract public function price();
+    }
+    
+As you can see, ``DeliveryFactory`` doesn't specify the exact class of the object that will be created.
+    
+Next, use settings similar to those in the sections above. These settings allow you to define a factory method for subclasses without explicitly extending the abstract class (i.e., without ``class DHL extends DeliveryFactory``)!
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/services.yaml
+        services:
+            # ...
+
+            App\Deliveries\DHL:
+                factory:   ['@App\Factories\DeliveryFactory', create]
+                arguments:
+                    $deliveryMethod: 'App\Deliveries\DHL'
```

Comment:
Let’s not use the variable key binding here, but:

arguments: [...]

Instead.

Same for xml and php

Reply:
Done.